### PR TITLE
CI: print logs from the cluster as a build step

### PR DESCRIPTION
**Pull request by piodul**
Some test failures may be caused due to issues in Cassandra or Scylla and it is hard to identify them if they are silent during the whole CI run.

This PR modifies our CI jobs that use docker-compose so that the logs from the cluster are always printed.

## Pre-review checklist

- [ ] I have split my patch into logically separate commits.
- [ ] All commit messages clearly explain what they change and why.
- [ ] I added relevant tests for new features and bug fixes.
- [ ] All commits compile, pass static checks and pass test.
- [ ] PR description sums up the changes and reasons why they should be introduced.
- [ ] I added appropriate `Fixes:` annotations to PR description.

### CODE REVIEW

**Comment by piodul** [comment]
It looks like printing the logs after failure works correctly, e.g. here: https://github.com/scylladb/scylla-rust-driver/actions/runs/4204000245/jobs/7294090832

File: .github/workflows/cassandra.yml [hunk 1]
```diff
@@ -24,5 +24,11 @@ jobs:
     - name: Run tests on cassandra
       run: |
         CDC='disabled' SCYLLA_URI=172.42.0.2:9042 SCYLLA_URI2=172.42.0.3:9042 SCYLLA_URI3=172.42.0.4:9042 cargo test --verbose -- --skip test_views_in_schema_info
-    - name: Stop cluster
+    - name: Stop the cluster
+      if: ${{ always() }}
+      run: docker-compose -f test/cluster/cassandra/docker-compose.yml stop
```
Comment (havaker) [inline .github/workflows/cassandra.yml:29]: Why did you add the `stop` command? Was this to see all the cluster logs, including those produced during Scylla shutdowns?

Comment (piodul) [reply .github/workflows/cassandra.yml:29]: Yes. I can't do it after `docker-compose shutdown` because it deletes the containers along with the logs.

**Review by havaker — APPROVED** [review]

**Comment by piodul** [comment]
@cvybhu ping

**Review by cvybhu — APPROVED** [review]
